File: beginner/pagina09/2630/2630.cpp
#include <iostream>
#include <sstream>
#include <iomanip>
#include <algorithm>

#define R 0.30
#define G 0.59
#define B 0.11

#define grayscale(r, g, b)(static_cast<int>(R * r + G * g + B * b))
#define minscale(r, g, b)( min(r, min(g, b)))
#define maxscale(r, g, b)( max(r, max(g, b)))
#define media(r, g, b)((r + g + b) / 3)

using namespace std;

void OUTPUT(int caso, int valor){
    cout << "Caso #" << caso << ": " << valor << endl;
}

int main(){

    int numero_casos;
    cin >> numero_casos;

    for(int i = 1; i <= numero_casos; i++){
        string tipo_conversao;
        int r, g, b, resultado;

        cin >> tipo_conversao >> r >> g >> b;

        if(tipo_conversao == "eye"){ resultado = grayscale(r, g, b); }
        if(tipo_conversao == "min"){ resultado = minscale(r, g, b); }
        if(tipo_conversao == "max"){ resultado = maxscale(r, g, b); }
        if(tipo_conversao == "mean"){ resultado = media(r, g, b); }

        OUTPUT(i, resultado);
    }
    return 0;
}

File: beginner/pagina09/2630/2630.js
const fs = require("fs")
const input = fs.readFileSync("/dev/stdin", "utf8").split("\n")

const R = 0.30;
const G = 0.59;
const B = 0.11;

const OUTPUT = (caso, valor) => console.log(`Caso #${caso}: ${valor}`)
const grayscale = (r, g, b) => Math.trunc(R * r + G * g + B * b);
const minscale = (r, g, b) => Math.min(r, g, b);
const maxscale = (r, g, b) => Math.max(r, g, b);
const average = (r, g, b) => Math.trunc((r + g + b) / 3);

const number_cases =  Number(input.shift())
 
for(let i = 1; i <= number_cases; i++){
    const type = input.shift()
    const [r, g, b] = input.shift().split(' ').map(Number)
    let resultado = 0;

    if(type == "eye"){ resultado = grayscale(r, g, b); }
    if(type == "min"){ resultado = minscale(r, g, b); }
    if(type == "max"){ resultado = maxscale(r, g, b); }
    if(type == "mean"){ resultado = average(r, g, b); }

    OUTPUT(i, resultado)
}